infrastructure/modules/container-apps/alerts.tf

```diff
@@ -36,3 +36,56 @@ resource "azurerm_monitor_scheduled_query_rules_alert_v2" "failure_event" {
     }
   }
 }
+
+# IMPORTANT:
+# Enable metrics store with all dimensions: https://docs.azure.cn/en-us/azure-monitor/app/metrics-overview?tabs=standard#custom-metrics-dimensions-and-preaggregation
+# currently this feature is in preview.
+resource "azurerm_monitor_scheduled_query_rules_alert_v2" "queue_length_high" {
+  for_each = var.enable_alerting ? toset([
+    "notifications-message-status-updates",
+    "notifications-message-batch-retries"
+  ]) : []
+
+  name                = "${var.app_short_name}-${each.key}-${var.environment}-queue-length-high-alert"
+  location            = var.region
+  resource_group_name = azurerm_resource_group.main.name
+
+  auto_mitigation_enabled = true
+  description             = "Alert when queue length exceeds ${var.queue_length_alert_threshold}"
+  display_name            = "${var.app_short_name} Notifications Queue Length High Alert"
+  enabled                 = true
```

Comment:
will we conditionally enable this like we do for all other `enable_alerting` gated things?